include/mbedtls/ssl.h

```diff
@@ -3120,6 +3188,73 @@ int mbedtls_ssl_conf_alpn_protocols( mbedtls_ssl_config *conf, const char **prot
 const char *mbedtls_ssl_get_alpn_protocol( const mbedtls_ssl_context *ssl );
 #endif /* MBEDTLS_SSL_ALPN */
 
+#if defined(MBEDTLS_SSL_DTLS_SRTP)
+/**
+ * \brief                   Add support for mki value in use_srtp extension.
+ *                          The default value is
+ *                          #MBEDTLS_SSL_DTLS_SRTP_MKI_UNSUPPORTED.
+ *
+ * \param conf              SSL configuration
+ * \param support_mki_value Enable or disable mki usage. Values are
+ *                          #MBEDTLS_SSL_DTLS_SRTP_MKI_UNSUPPORTED
+ *                          or #MBEDTLS_SSL_DTLS_SRTP_MKI_SUPPORTED.
+ */
+void mbedtls_ssl_conf_srtp_mki_value_supported( mbedtls_ssl_config *conf,
+                                                int support_mki_value );
+
+/**
+ * \brief                   Set the supported DTLS-SRTP protection profiles.
+ *
+ * \param conf              SSL configuration
+ * \param profiles          List of supported protection profiles,
+ *                          in decreasing preference order.
+ * \param profiles_number   Number of supported profiles.
+ *
+ * \return                  0 on success, or #MBEDTLS_ERR_SSL_BAD_INPUT_DATA.
+ */
+int mbedtls_ssl_conf_dtls_srtp_protection_profiles
+                               ( mbedtls_ssl_config *conf,
+                                 const mbedtls_ssl_srtp_profile *profiles,
```

Comment:
The parameter is an array of mbedtls_ssl_srtp_profile which is an enum, so a compiler would not accept a 0 value.
But MBEDTLS_SRTP_UNSET_PROFILE shall be mapped to 0. I can switch to a MBEDTLS_SRTP_UNSET_PROFILE terminated array for the profiles parameter and remove the profiles_number parameter.

Reply:
Generally I prefer pointer+length to a zero-terminated array because it's easier to keep track of the buffer size. With a length: allocate N elements, pass N together with the pointer everywhere. With zero termination, you have to remember to create the trailing zero, and if the array isn't constant you also have to ensure that the creation doesn't overflow the array.

However consistency with existing zero-terminated lists in the interface is a valid argument. So I don't have strong feelings on this one.

Reply:
@hanno-arm  any comment? Shall I let it the way it is or change for the MBEDTLS_SRTP_UNSET_PROFILE-terminated list?

Reply:
I'd prefer an `MBEDTLS_SRTP_UNSET_PROFILE`-terminated list to be aligned with the rest of the API.